reader/reader_test.go

```diff
@@ -0,0 +1,36 @@
+package reader
+
+import (
+	"log"
+	"os"
+	"testing"
+)
+
+func TestReadCsv(t *testing.T) {
+	name := "test.csv"
+	f := createFile(name)
```

Comment:
There is no need for this name. Try this:
```go
	f, err := createFile(tcs)
	if err != nil {
		t.Fatalf("writing file: %v", err)
	}
	defer f.Close()
```
We are doing the error check here. The createFile is only a helper and doesn't have to fail.
I will leave comments line by line, and at the end of this file I will leave the whole changes, just read on.